lib/cocoapods-core/specification/consumer.rb

```diff
@@ -324,7 +329,11 @@ def merge_values(attr, existing_value, new_value)
       #
       def prepare_value(attr, value)
         if attr.container == Array
-          value = [*value].compact
+          value = if value.is_a?(Hash)
```

Comment:
@segiddins this is the primary fix that helps me accomplish this. Basically I want to the container to be an "Array" for this attribute since I want to preserve the order in which the script phases are added.

However, without this change the Hash becomes an array and then I lose the ability to recreate the hashes from the arrays.

This is the first attribute that requires an array of hashes so this is probably why I hit this.